WordPress/gutenberg
# Mobile: Enhance the redux mobile enhancer to use the MediaQueries API

For more performance

This PR updates the Redux Mobile Enhancer to use `window.matchMedia` which matches `redux-responsive` behavior. As it's more performant than relying on the browser's `onResize` event.

**Testing instructions**

 - Check that the mobile sidebar state is not tied to the desktop sidebar state (resize and toggle the sidebar multiple times)
